access.lisp

```diff
@@ -524,6 +528,19 @@
         (actual-slot-name
          (set-access new o actual-slot-name))
         ))
+    o)
+
+  (:method (new (o structure-object) k &key type test key)
+    (declare (ignore type test key))
+    (let ((actual-slot-name (has-slot? o k)))
```

Comment:
You mustn't use `has-slot?` here since even calling it here probably results in UB. There is no portable way to check if a structure contains a slot with a given name and using MOP wizardry on structure objects is generally undefined. The keyword argument `lax?` should either signal an error or be ignored altogether when used on structure objects.

Reply:
Actually, I'm wrong. `(progn (defstruct foo (bar 42)) (slot-exists-p (make-foo) 'bar))` seems to work on SBCL, CCL, ECL, CLISP, ABCL, Clasp, LispWorks, and ACL. Maybe Access should make use of this function?

Reply:
Ok. I'll have look. How do you know it works for all those implementations? By looking at the docs?

Reply:
Nope, I just executed this test on all implementations and in all cases a true value was returned. Maybe you can write a slightly more contrived test - for `(defstruct foo (bar 42))`, `(slot-exists-p (make-foo) 'bar)` must return true and `(slot-exists-p (make-foo) 'baz)` must return false.

Reply:
Ok. I wonder how you do that, as I'm very slow when trying the different implementations :)

Reply:
I have SBCL, CCL, ABCL, ECL, CLISP, Clasp, LW, and ACL installed locally and, except for LW Personal, it's a matter of typing in the implementation name, pasting the code snippet, checking if `42` was returned, hitting CTRL+D to exit, and repeating.

Reply:
I don't understand the lax semantics here. Maybe you can write this part?

Reply:
...truth be told, I don't understand it either, and I don't know if I can have enough focus nowadays to be able to understand this behavior or even rationalize it (e.g. it will likely do the wrong thing in case of slots with names that are `string=` but not `eq`).

Reply:
The lax semantics are a match by string-equal the symbol name of each of the keys (EG:  `(has-slot? o "MyKeyName")` will return ` 'mykeyname` from whatever package was used when creating the slot).  The goal here was to compare key names without packages as this is often confusing to beginner's and instead to treat all key names as case-insensitive strings regardless of their specification.  `has-slot?` always returns the actual slot name regardless of the slot-name passed in.  

If the slot-name passed in matches multiple slots on the object (an indication that you should pass the exact symbol instead), access will emit a warning and return the first slot it found.

Does this clear up the confusion?

Reply:
That clears it up. Thanks. I'll have another look at the code again with that in mind.